numpy/lib/tests/test_shape_base.py

```diff
@@ -466,6 +569,46 @@ def test_unequal_split(self):
         a = np.arange(10)
         assert_raises(ValueError, split, a, 3)
 
+    def test_equal_two_dimensional_split(self):
+        matrix = np.reshape(np.arange(16), (4, 4))
+
+        res = split(matrix, [4, 2], (0, 1))
+        desired = np.empty(8, dtype=object)
+        desired[0] = np.array([[0, 1]])
+        desired[1] = np.array([[2, 3]])
+        desired[2] = np.array([[4, 5]])
+        desired[3] = np.array([[6, 7]])
+        desired[4] = np.array([[8, 9]])
+        desired[5] = np.array([[10, 11]])
+        desired[6] = np.array([[12, 13]])
+        desired[7] = np.array([[14, 15]])
+        desired = np.reshape(desired, (4, 2))
+
+        assert(res.shape == desired.shape)
+
+        for i, element in enumerate(res):
+            compare_results(element, desired[i])
+
+        res = split(matrix, [2, 2], (0, 1))
+        desired = np.empty(4, dtype=object)
+        desired[0] = np.array([[0, 1], [4, 5]])
+        desired[1] = np.array([[2, 3], [6, 7]])
+        desired[2] = np.array([[8, 9], [12, 13]])
+        desired[3] = np.array([[10, 11], [14, 15]])
+        desired = np.reshape(desired, (2, 2))
+
+        assert (res.shape == desired.shape)
+
+        for i, element in enumerate(res):
+            compare_results(element, desired[i])
+
+    def test_unequal_two_dimensional_split(self):
+        matrix = np.reshape(np.arange(16), (4, 4))
+        assert_raises(ValueError, split, matrix, [3, 2], (0, 1))
+        assert_raises(ValueError, split, matrix, [2, 3], (0, 1))
+        assert_raises(ValueError, split, matrix, [2], (0, 1))
+        assert_raises(ValueError, split, matrix, 3, (0, 1))
+
```

Comment:
Can you add the trivial case,
```suggestion
    
    def test_zero_dimensional_split(self):
        matrix = np.reshape(np.arange(16), (4, 4))
        
        ret = split(matrix, [], axis=())
        assert_equal(ret.ndim, 0)
        assert_equal(ret[()], matrix)
        
        # too many split arguments
        assert_raises(ValueError, split, matrix, [3], axis=())
```